src/System.Net.HttpListener/src/System/Net/Managed/HttpEndPointManager.cs

```diff
@@ -75,6 +75,26 @@ public static void AddPrefix(string prefix, HttpListener listener)
 
         private static void AddPrefixInternal(string p, HttpListener listener)
         {
+            try
+            {
+                int start = p.IndexOf(':') + 3;
+                int colon = p.IndexOf(':', start, p.Length - start);
+                if (colon != -1)
+                {
+                    // root can't be -1 here, since we've already checked for ending '/' in ListenerPrefix.
+                    int root = p.IndexOf('/', colon, p.Length - colon);
+                    int port = int.Parse(p.Substring(colon + 1, root - colon - 1));
+                    if (port <= 0 || port >= 65536)
+                    {
+                        throw new HttpListenerException((int)HttpStatusCode.BadRequest, SR.net_invalid_port);
+                    }
+                }
+            }
+            catch
+            {
+                throw new HttpListenerException((int)HttpStatusCode.BadRequest, SR.net_invalid_port);
+            }
```

Comment:
It'd be better to write the code in the try so that this catch block isn't needed, i.e. validate the input along the way, use TryParse instead of Parse, check the result of the IndexOf calls, etc., and throw an appropriate exception then, rather than hoping that int.Parse throws or p.Substring throws or whatever, catching everything, and then throwing another exception.